contracts/RMRK/utils/RMRKBulkWriter.sol

```diff
@@ -216,6 +224,126 @@ contract RMRKBulkWriter is Context {
+     * @dev If the `equipData` is set, the NFT will be equipped after the transfer. Mind that the child index equip data is the index after the transfer, so it MUST be set to the number of children the destination NFT has.
+     * @dev If equipping into destination, the `equipData.tokenId` will match the `destinationTokenId`.
+     * @dev This contract MUST have approval on the parent collection, if any, to transfer the NFT.
+     * @dev This contract MUST have approval for assets on the parent collection, if any, to unequip the NFT.
+     * @dev This contract MUST have approval on the NFT collection, if not nested, to transfer the NFT.
+     * @dev This contract MUST have approval on the destination collection to accept the NFT.
+     * @dev This contract MUST have approval for assets on the destination collection to equip the NFT.
+     * @param collection Address of the collection that this contract is managing
+     * @param tokenId ID of the token we are managing
+     * @param destinationCollection Address of the destination collection
+     * @param destinationTokenId ID of the destination token
+     * @param parentData A `ParentData` struct specifying the parent data
+     * @param equipData An `IntakeEquip` struct specifying the equip data
+     */
+    function transferAndEquip(
+        address collection,
+        uint256 tokenId,
+        address destinationCollection,
+        uint256 destinationTokenId,
+        ParentData memory parentData,
+        IERC6220.IntakeEquip memory equipData
+    ) public onlyTokenOwner(collection, tokenId) {
+        if (parentData.parentCollection != address(0)) {
+            if (
+                parentData.parentEquippedAssetId != 0 &&
+                parentData.parentEquippedSlotPartId != 0
+            ) {
+                IERC6220(parentData.parentCollection).unequip(
+                    parentData.parentTokenId,
+                    parentData.parentEquippedAssetId,
+                    parentData.parentEquippedSlotPartId
+                );
+            }
+            IERC7401(parentData.parentCollection).transferChild(
+                parentData.parentTokenId,
+                destinationCollection,
+                destinationTokenId,
+                parentData.childIndexInParent,
+                collection,
+                tokenId,
+                false,
+                ""
+            );
+        } else {
+            IERC7401(collection).nestTransferFrom(
+                _msgSender(),
+                destinationCollection,
+                tokenId,
+                destinationTokenId,
+                ""
+            );
+        }
+
+        _acceptChildIfNecessary(
+            destinationCollection,
+            destinationTokenId,
+            collection,
+            tokenId
+        );
+
+        if (equipData.assetId == 0) return;
+
+        _unequipSlotIfNecessary(
```

Comment:
wondering if there's any benefit in re-using replaceEquipement function from this contract here instead?

Reply:
Sound great, will give it a try.